esiclient/v1/switch.py

```diff
@@ -172,3 +171,284 @@ def take_action(self, parsed_args):
         print("Disabling access to {0}".format(switchport))
 
         ironic_client.node.vif_detach(port.node_uuid, np_uuid)
+
+
+class EnableTrunkPort(command.ShowOne):
+    """Configure Switchport Trunk Access"""
+
+    log = logging.getLogger(__name__ + ".EnableTrunkPort")
+
+    def get_parser(self, prog_name):
+        parser = super(EnableTrunkPort, self).get_parser(prog_name)
+        parser.add_argument(
+            "switch",
+            metavar="<switch>",
+            help=_("Switch"))
+        parser.add_argument(
+            "switchport",
+            metavar="<switchport>",
+            help=_("Switchport"))
+        parser.add_argument(
+            "vlan_id",
+            metavar="<vlan_id>",
+            help=_("VLAN ID"))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        switch = parsed_args.switch
+        switchport = parsed_args.switchport
+        vlan_id = parsed_args.vlan_id
+
+        # get associated port and node
+        ironic_client = self.app.client_manager.baremetal
+        ports = ironic_client.port.list(detail=True)
+        port = next((port for port in ports
+                     if port.local_link_connection.get(
+                             'port_id') == switchport and
+                     port.local_link_connection.get(
+                         'switch_info') == switch),
+                    None)
+        if not port:
+            raise exceptions.CommandError("ERROR: Switchport unknown")
+        node = ironic_client.node.get(port.node_uuid)
+
+        # get associated network
+        neutron_client = self.app.client_manager.network
+        networks = list(neutron_client.networks(
+            provider_network_type='vlan',
+            provider_segmentation_id=vlan_id))
+        network = next(iter(networks), None)
+        if not network:
+            raise exceptions.CommandError("ERROR: VLAN ID unknown")
+
+        # create trunk
+        trunk_name = switch + switchport
```

Comment:
should it be switch +  '-'  + switchport?

Reply:
Good point! Fixed here (and other places)